vignettes/Example1-Ostats-univariate.Rmd

```diff
@@ -220,13 +222,121 @@ Ostats_plot(plots = siteID,
+
+```{r out.width="80%", warning=FALSE, error=FALSE, fig.align="center", fig.height=4.5, fig.width=7}
+siteID <- factor(ant_data$chamber)
+taxonID <- factor(ant_data$species)
+trait <-  as.matrix(ant_data[, 'time', drop = FALSE])
+
+
+Ostats_plot(plots = siteID, 
+            sp = taxonID, 
+            traits = trait, 
+            overlap_dat = circular_example, 
+            name_x = 'Time (hours)', 
+            circular = TRUE,
+            discrete = TRUE)
+```
+
+
+## Boostrapping
+
+### Example for generating bootstrap of linear univariate O-statistics
+
+This is an illustration of how to bootstrap O-statistics using the example data on small mammal body mass from Harvard Forest (HARV) and Jornada (JORN). It is written without any tidyverse packages, other than ggplot2, to not add any new dependencies.
+
+The code to load a small example dataset (two sites of NEON small mammal body weights) is copied from the above in the vignette in case users are working through examples separately.
+
+```{r}
+library(Ostats)
+library(ggplot2)
+
+### Code copied from vignette to load a small example dataset and calculate O-stats on the observed dataset.
+dat <- small_mammal_data[small_mammal_data$siteID %in% c('HARV', 'JORN'),
+                         c('siteID', 'taxonID', 'mass')]
+dat <- dat[!is.na(dat$mass), ]
+dat$log_mass <- log10(dat$mass)
+```
+
+Next, calculate O-stats on the dataset (don't do the null model, just calculate the observed overlap statistic). Then pull the normalized overlap out into a data frame for plotting later.
+
+```{r}
+Ostats_example <- Ostats(traits = as.matrix(dat[,'log_mass', drop = FALSE]),
+                         sp = factor(dat$taxonID),
+                         plots = factor(dat$siteID),
+                         run_null_model = FALSE)
+
+# Convert observed normalized overlap to a data frame for plotting
+Ostats_observed <- with(Ostats_example, data.frame(siteID = dimnames(overlaps_norm)[[1]], overlap_norm = round(overlaps_norm[,1], 4)))
+```
+
+Define a custom function to do the bootstrap. Following the suggestion of Maitner et al. (2023), we sample with replacement from each species' traits within each community proportional to that species' relative abundance in that community. We use the nonparametric bootstrap approach. In other words, we are sampling directly from the raw data, instead of fitting a distribution and sampling from that, which would be parametric because we would be using the parameters of a fitted distribution to generate bootstrap samples.
+
+```{r}
+boot_fn <- function(i) {
+  # Bootstrap sample of same size as the data, within each community each row is sampled with replacement with equal probability
+  # This means the proportion of individuals of each species in the bootstrap sample will be similar to the observed data.
+  # Commented out version uses dplyr, the uncommented version uses base R code to get the same result.
```

Comment:
Lines 292-296 here should be deleted. We do not need them anymore, it was only my first attempt of writing this using dplyr code before I later went back and rewrote it using no external packages.

Reply:
Thanks for flagging this - I didn't catch it before.